edx_solutions_api_integration/users/views.py

```diff
@@ -1413,38 +1413,28 @@ def get(self, request, *args, **kwargs):  # pylint: disable=unused-arguments
             cache_course_user_data('social', course_id, user.id, {'score': social_engagement_score})
         else:
             data = cached_social_data
+
         if include_stats:
-            data['stats'] = self._get_user_discussion_metrics(request, user, course_id)
+            cached_social_stats = get_cached_data('social_stats', course_id, user.id)
```

Comment:
This will never work because we aren't actually caching it in the first place. `if not cached_social_stats`, then after getting the stats we need to `cache_course_user_data` as done above for the `'social'` key, but now for `'social_stats'`.

Reply:
Ah, that's right, I didn't set the cache here.

I'm not sure whether we need it to be up-to-date. I'm actually not sure why it's cached this way right now - `social_engagement_score` has TTL of 1h and `course_avg` has TTL of 30m.

Reply:
I guess if the admins check this stuff often it may be useful to cache, otherwise I think there're many database calls involved.